pandas/core/groupby/ops.py

```diff
@@ -859,7 +859,21 @@ def ngroups(self) -> int:
     def reconstructed_codes(self) -> list[np.ndarray]:
         codes = self.codes
         ids, obs_ids, _ = self.group_info
-        return decons_obs_group_ids(ids, obs_ids, self.shape, codes, xnull=True)
+        reconstructed_codes = decons_obs_group_ids(
```

Comment:
umm would be better to actually do this in `decons_obs_group_ids`

Reply:
i dunno, not wild about core.sorting needing to know anything about self._groupings

Reply:
hmm i take this back, this now makes sorting dependent on groupby.ops which is very strange.

@CloseChoice so instead am ok to pass *something* do `decons_obs_group_ids` but the something should not be a list of grouping, rather list of codes or similar

Reply:
changed this to pass only a list where we need to reconstruct the codes for na